### ISSUE TITLE
About finetune on unidepthv2

### ISSUE THREAD

**Issue by wuqun-tju** [lpiccinelli-eth/UniDepth#117]
Hello
 
Thank you for your great work

I want to finetune on my own data. I have some question

1. I found there no get_pretrain() func on decoder module
2. I found in warm up step , the lr is constant because base_value is equal to init_value, is it right in you own training?

![Image](https://github.com/user-attachments/assets/97d9ad9a-788a-4547-9651-371529cfb67c)

3. then I found some loss is up and some loss is constant, dou you have any advice

![Image](https://github.com/user-attachments/assets/492f718a-5947-4a1a-85e8-1fe9cd7e8e9f)

I am lookingforward your reply, Thank you very much

**Comment by lpiccinelli-eth** [comment]
Hey, thanks a lot for your questions!

1. If you're referring to get_params, they are obtained via the [get_params](https://github.com/lpiccinelli-eth/UniDepth/blob/8f44e04da58c00c3d1793099ed8000445d5363dc/unidepth/models/unidepthv2/unidepthv2.py#L409) function here.
2. We actually never used a warmup. The relevant code is a leftover from earlier experiments—it’s more of "legacy dirtiness" from when I tried warmup, but I didn’t observe any consistent benefit.
3. I'm a bit confused by the losses going to zero while the SiLog increases significantly. Regarding the camera's Regression, what you posted is correct. During training, we used the snippet under the comment `LEGACY CODE FOR TRAINING`, where we randomly sample between GT and predicted rays using a sort of curriculum learning with `prob = 0.8 * (1 - tanh(...)) + 0.2`.
4. The +2 offset is just to improve the initialization at the beginning of training. Initially, the decoder outputs log-depth values close to zero, so adding +2 shifts the depth to around exp(2), which is a typical average depth value across both indoor and outdoor scenes. This helps avoid wasting capacity or causing large gradients on the final layer’s bias just to reach a reasonable average depth.
5. You can find the random translation [here](https://github.com/lpiccinelli-eth/UniDepth/blob/8f44e04da58c00c3d1793099ed8000445d5363dc/unidepth/datasets/pipelines/transforms.py#L866). You're right that this class should be updated to use the new camera abstraction instead of hardcoded intrinsics—I’ll fix that!

Let me know if you have any more questions!

**Comment by dxw2000** [comment]
> Hello  你好
> 
> Thank you for your great work感谢您的出色工作
> 
> I want to finetune on my own data. I have some question我想对自己的数据进行微调。我有一些问题
> 
> 1. I found there no get_pretrain() func on decoder module我发现解码器模块上没有 get_pretrain（） 功能
> 2. I found in warm up step , the lr is constant because base_value is equal to init_value, is it right in you own training?我在热身步骤中发现，lr 是恒定的，因为 base_value 等于 init_value，这在你自己的训练中正确吗？
> 
> ![Image](https://github.com/user-attachments/assets/97d9ad9a-788a-4547-9651-371529cfb67c)
> 
> 3. then I found some loss is up and some loss is constant, dou you have any advice然后我发现有些损失是上升的，有些损失是恒定的，你有什么建议吗
> 
> ![Image](https://github.com/user-attachments/assets/492f718a-5947-4a1a-85e8-1fe9cd7e8e9f)
> 
> I found regression loss is constant is that it used rays_gt not rays_pred, is it right?我发现回归损失是恒定的，因为它 rays_gt 使用而不是 rays_pred，对吗？
> 
> ![Image](https://github.com/user-attachments/assets/7f583752-2b30-4f1c-9aab-44eb9c9f9dae) 4. why logdepth plus 2.0 in decoder4. 为什么解码器中使用 LogDepth Plus 2.0
> 
> ![Image](https://github.com/user-attachments/assets/70ab953b-9abb-428e-96d9-6af12b4bff7b) 5. I haven't find relative translation t-u[-0.1,0.1] in the code as described as follows:5. 我在代码中没有找到相对翻译 t-u[-0.1,0.1]，如下所述：
> 
> ![Image](https://github.com/user-attachments/assets/48b7f036-81d5-48e0-b90c-91447b0fb052)
> 
> I am lookingforward your reply, Thank you very much期待您的回复，非常感谢

Could you please share how to use a custom dataset for fine-tuning and the methods for generating the data? Thank you very much.

**Comment by lixueqidaytoy** [comment]
您好，请问您后续有解决相关的问题吗？我也是在自己的数据集上微调，出现了和您相似的问题。我在训练时采用了预测的ray（也就是使用注释掉的那几行），SIlog损失也是会突然增加到1万多，模型收敛十分困难。我的batchsize设置的是16，这会和batchsize太小有关吗？会和自己的数据集的深度太稀疏有关吗？

<img width="1145" height="149" alt="Image" src="https://github.com/user-attachments/assets/aae0eae6-5b3f-48ed-a603-88866365993b" />

<img width="807" height="153" alt="Image" src="https://github.com/user-attachments/assets/a897b457-922d-47e9-8af1-301ee9f91d4d" />

Hello, have you managed to resolve this issue later? I'm also fine-tuning on my own dataset and encountered a similar problem. During training, I used predicted rays (i.e., using the lines that were commented out), and the SIlog loss also suddenly spikes to over 10,000, making model convergence extremely difficult. My batch size is set to 16. Could this be related to the batch size being too small? Or could it be related to the depth maps in my dataset being too sparse?

**Comment by Cats-eat-peppercorns** [comment]
@dxw2000 请问你是否成功微调了模型，我目前微调nyu的vitl模型但d1始终达不到98.8%的精度，我只有93%。我的配置如下：
`{
    "generic": {
        "seed": 13,
        "deterministic": true
    },
    "training": {
        "n_iters": 15000,
        "batch_size": 8,
        "validation_interval": 500,
        "nsteps_accumulation_gradient": 2,
        "use_checkpoint": false,
        "find_unused_parameters": false,

        "lr": 1e-4,
        "lr_final": 1e-6,
        "lr_warmup": 0.02,
        "cycle_beta": false,

        "pretrained": "/home/xzy/weights/pytorch_model_l.bin",

        "wd": 0.1,
        "wd_final": 0.1,
        "warmup_iters": 2000,
        "ld": 1.0,

        "drop_path": 0.0,
        "ema": true,
        "f16": true,
        "clipping": 1.0,

        "losses": {
            "depth": {
                "name": "SILog",
                "weight": 1.0,
                "output_fn": "sqrt",
                "input_fn": "log",
                "dims": [-2, -1],
                "integrated": 0.15
            },
            "invariance": {
                "name": "SelfDistill",
                "weight": 0.1,
                "output_fn": "sqrt"
            },
            "camera": {
                "name": "Regression",
                "weight": 0.25,
                "gamma": 1.0,
                "alpha": 1.0,
                "fn": "l2",
                "output_fn": "sqrt",
                "input_fn": "linear"
            },
            "ssi": {
                "name": "EdgeGuidedLocalSSI",
                "weight": 1.0,
                "output_fn": "sqrt",
                "input_fn": "log1i",
                "use_global": true,
                "min_samples": 6
            },
            "confidence": {
                "name": "Confidence",
                "weight": 0.1,
                "gamma": 1.0,
                "alpha": 1.0,
                "fn": "l1",
                "output_fn": "sqrt",
                "input_fn": "linear"
            }
        }
    },
    "data": {
        "image_shape": [480, 640],
        "normalization": "imagenet",
        "num_copies": 2,
        "num_frames": 1,
        "sampling": {
            "NYUv2Depth": 1.0
        },
        "train_datasets": [
            "NYUv2Depth"
        ],
        "val_datasets": [
            "NYUv2Depth"
        ],
        "data_root": "/home/xzy/datasets",
        "crop": "garg",
        "augmentations": {
            "random_scale": 2.0,
            "random_jitter": 0.4,
            "jitter_p": 0.8,
            "random_blur": 2.0,
            "blur_p": 0.2,
            "random_gamma": 0.2,
            "gamma_p": 0.8,
            "grayscale_p": 0.2,
            "flip_p": 0.5,
            "test_context": 1.0,
            "shape_constraints": {
                "ratio_bounds": [0.5, 2.5],
                "pixels_max": 600000,
                "pixels_min": 200000,
                "height_min": 15,
                "width_min": 15,
                "shape_mult": 14,
                "sample": true
            }
        }
    },
    "model": {
        "name": "UniDepthV2",
        "num_heads": 8,
        "expansion": 4,
        "layer_scale": 1.0,
        "pixel_decoder": {
            "name": "Decoder",
            "hidden_dim": 512,
            "dropout": 0.0,
            "depths": [2, 2, 2],
            "out_dim": 64,
            "kernel_size": 3
        },
        "pixel_encoder": {
            "lr": 2e-6,
            "wd": 0.1,
            "name": "dinov2_vitl14",
            "frozen_stages": 0,
            "num_register_tokens": 0,
            "use_norm": true,
            "freeze_norm": true,
            "pretrained": "/home/xzy/weights/dinov2_vitl14_pretrain.pth",
            "stacking_fn": "last",
            "output_idx": [6, 12, 18, 24]
        }
    }
}
`

**Comment by 999-jpg** [comment]
> [@dxw2000](https://github.com/dxw2000) 请问你是否成功微调了模型，我目前微调nyu的vitl模型但d1始终达不到98.8%的精度，我只有93%。我的配置如下： `{ "generic": { "seed": 13, "deterministic": true }, "training": { "n_iters": 15000, "batch_size": 8, "validation_interval": 500, "nsteps_accumulation_gradient": 2, "use_checkpoint": false, "find_unused_parameters": false,
> 
> ```
>     "lr": 1e-4,
>     "lr_final": 1e-6,
>     "lr_warmup": 0.02,
>     "cycle_beta": false,
> 
>     "pretrained": "/home/xzy/weights/pytorch_model_l.bin",
> 
>     "wd": 0.1,
>     "wd_final": 0.1,
>     "warmup_iters": 2000,
>     "ld": 1.0,
> 
>     "drop_path": 0.0,
>     "ema": true,
>     "f16": true,
>     "clipping": 1.0,
> 
>     "losses": {
>         "depth": {
>             "name": "SILog",
>             "weight": 1.0,
>             "output_fn": "sqrt",
>             "input_fn": "log",
>             "dims": [-2, -1],
>             "integrated": 0.15
>         },
>         "invariance": {
>             "name": "SelfDistill",
>             "weight": 0.1,
>             "output_fn": "sqrt"
>         },
>         "camera": {
>             "name": "Regression",
>             "weight": 0.25,
>             "gamma": 1.0,
>             "alpha": 1.0,
>             "fn": "l2",
>             "output_fn": "sqrt",
>             "input_fn": "linear"
>         },
>         "ssi": {
>             "name": "EdgeGuidedLocalSSI",
>             "weight": 1.0,
>             "output_fn": "sqrt",
>             "input_fn": "log1i",
>             "use_global": true,
>             "min_samples": 6
>         },
>         "confidence": {
>             "name": "Confidence",
>             "weight": 0.1,
>             "gamma": 1.0,
>             "alpha": 1.0,
>             "fn": "l1",
>             "output_fn": "sqrt",
>             "input_fn": "linear"
>         }
>     }
> },
> "data": {
>     "image_shape": [480, 640],
>     "normalization": "imagenet",
>     "num_copies": 2,
>     "num_frames": 1,
>     "sampling": {
>         "NYUv2Depth": 1.0
>     },
>     "train_datasets": [
>         "NYUv2Depth"
>     ],
>     "val_datasets": [
>         "NYUv2Depth"
>     ],
>     "data_root": "/home/xzy/datasets",
>     "crop": "garg",
>     "augmentations": {
>         "random_scale": 2.0,
>         "random_jitter": 0.4,
>         "jitter_p": 0.8,
>         "random_blur": 2.0,
>         "blur_p": 0.2,
>         "random_gamma": 0.2,
>         "gamma_p": 0.8,
>         "grayscale_p": 0.2,
>         "flip_p": 0.5,
>         "test_context": 1.0,
>         "shape_constraints": {
>             "ratio_bounds": [0.5, 2.5],
>             "pixels_max": 600000,
>             "pixels_min": 200000,
>             "height_min": 15,
>             "width_min": 15,
>             "shape_mult": 14,
>             "sample": true
>         }
>     }
> },
> "model": {
>     "name": "UniDepthV2",
>     "num_heads": 8,
>     "expansion": 4,
>     "layer_scale": 1.0,
>     "pixel_decoder": {
>         "name": "Decoder",
>         "hidden_dim": 512,
>         "dropout": 0.0,
>         "depths": [2, 2, 2],
>         "out_dim": 64,
>         "kernel_size": 3
>     },
>     "pixel_encoder": {
>         "lr": 2e-6,
>         "wd": 0.1,
>         "name": "dinov2_vitl14",
>         "frozen_stages": 0,
>         "num_register_tokens": 0,
>         "use_norm": true,
>         "freeze_norm": true,
>         "pretrained": "/home/xzy/weights/dinov2_vitl14_pretrain.pth",
>         "stacking_fn": "last",
>         "output_idx": [6, 12, 18, 24]
>     }
> }
> ```
> 
> } `



> 请问你是否成功微调了模型，我目前微调nyu的vitl模型但d1始终达不到98.8%的精度，我只有93%。我的配置如下： '{ “generic”： {  “通用的”： { “seed”： 13，  “种子”：13， “deterministic”： true  “确定性的”：真 }， “training”： {  “训练”： { “n_iters”： 15000，  "n_iters": 15000, “batch_size”： 8， “validation_interval”： 500，“validation_interval”：500， “nsteps_accumulation_gradient”： 2， “use_checkpoint”： false，“use_checkpoint”：false， “find_unused_parameters”： false，“find_unused_pa​​rameters”：false，
> 
> ```
>     "lr": 1e-4,
>     "lr_final": 1e-6,
>     "lr_warmup": 0.02,
>     "cycle_beta": false,
> 
>     "pretrained": "/home/xzy/weights/pytorch_model_l.bin",
> 
>     "wd": 0.1,
>     "wd_final": 0.1,
>     "warmup_iters": 2000,
>     "ld": 1.0,
> 
>     "drop_path": 0.0,
>     "ema": true,
>     "f16": true,
>     "clipping": 1.0,
> 
>     "losses": {
>         "depth": {
>             "name": "SILog",
>             "weight": 1.0,
>             "output_fn": "sqrt",
>             "input_fn": "log",
>             "dims": [-2, -1],
>             "integrated": 0.15
>         },
>         "invariance": {
>             "name": "SelfDistill",
>             "weight": 0.1,
>             "output_fn": "sqrt"
>         },
>         "camera": {
>             "name": "Regression",
>             "weight": 0.25,
>             "gamma": 1.0,
>             "alpha": 1.0,
>             "fn": "l2",
>             "output_fn": "sqrt",
>             "input_fn": "linear"
>         },
>         "ssi": {
>             "name": "EdgeGuidedLocalSSI",
>             "weight": 1.0,
>             "output_fn": "sqrt",
>             "input_fn": "log1i",
>             "use_global": true,
>             "min_samples": 6
>         },
>         "confidence": {
>             "name": "Confidence",
>             "weight": 0.1,
>             "gamma": 1.0,
>             "alpha": 1.0,
>             "fn": "l1",
>             "output_fn": "sqrt",
>             "input_fn": "linear"
>         }
>     }
> },
> "data": {
>     "image_shape": [480, 640],
>     "normalization": "imagenet",
>     "num_copies": 2,
>     "num_frames": 1,
>     "sampling": {
>         "NYUv2Depth": 1.0
>     },
>     "train_datasets": [
>         "NYUv2Depth"
>     ],
>     "val_datasets": [
>         "NYUv2Depth"
>     ],
>     "data_root": "/home/xzy/datasets",
>     "crop": "garg",
>     "augmentations": {
>         "random_scale": 2.0,
>         "random_jitter": 0.4,
>         "jitter_p": 0.8,
>         "random_blur": 2.0,
>         "blur_p": 0.2,
>         "random_gamma": 0.2,
>         "gamma_p": 0.8,
>         "grayscale_p": 0.2,
>         "flip_p": 0.5,
>         "test_context": 1.0,
>         "shape_constraints": {
>             "ratio_bounds": [0.5, 2.5],
>             "pixels_max": 600000,
>             "pixels_min": 200000,
>             "height_min": 15,
>             "width_min": 15,
>             "shape_mult": 14,
>             "sample": true
>         }
>     }
> },
> "model": {
>     "name": "UniDepthV2",
>     "num_heads": 8,
>     "expansion": 4,
>     "layer_scale": 1.0,
>     "pixel_decoder": {
>         "name": "Decoder",
>         "hidden_dim": 512,
>         "dropout": 0.0,
>         "depths": [2, 2, 2],
>         "out_dim": 64,
>         "kernel_size": 3
>     },
>     "pixel_encoder": {
>         "lr": 2e-6,
>         "wd": 0.1,
>         "name": "dinov2_vitl14",
>         "frozen_stages": 0,
>         "num_register_tokens": 0,
>         "use_norm": true,
>         "freeze_norm": true,
>         "pretrained": "/home/xzy/weights/dinov2_vitl14_pretrain.pth",
>         "stacking_fn": "last",
>         "output_idx": [6, 12, 18, 24]
>     }
> }
> ```
> 
> } `

请问你还在继续尝试吗，我看了一下我们的设置，除了我没有选择导入    "pretrained": "/home/xzy/weights/pytorch_model_l.bin"还有一些warmup相关的设置以外应该都是一样的（作者说warmup只是历史遗留代码），我的d1可以达到98.2%左右，但是AbsRel的值似乎有些异常偏高，我想知道你有遇到这样的状况吗

**Comment by Cats-eat-peppercorns** [comment]
@999-jpg 我的abs没有问题，现在我d1能到97.3%，但是还是到不了98.2%这么高。我能加你吗，我qq是1752857650，我想具体了解一下，你的数据集是自己做的吗？我自己做的数据集，但是感觉还是达不到你的精度。你记得@我，这样我才能看到你的回复，不然我看不到

**Comment by Cats-eat-peppercorns** [comment]
@999-jpg 朋友，邮箱联系或者加我vx，邮箱是XZY18135917615@163.com。我很需要一个能交流这篇论文的人呀，baseline都搞不出文章里的指标，我真的很晕。怀疑是数据集制作的问题，想交流一下。
